test/unit/handlers/validator_test.rb

```diff
@@ -82,8 +82,96 @@ class ValidatorTest < ActiveSupport::TestCase
         exception = assert_raises(Exceptions::InvalidHandler) do
           Validator.verify_handler_integrity(handler_hash)
         end
-        assert_equal "dummy_handler is not valid: [wazzzzzzaaa]: No regex was provided. No valid action was provided.",
-                     exception.message
+        assert_equal(
+          "dummy_handler is not valid: [wazzzzzzaaa]: No regex or base command was provided. No valid action was provided.",
+          exception.message
+        )
+      end
+
+      test "#verify_handler_integrity raises invalid regex error" do
+        handler_hash = {
+          "dummy_handler" => {
+            "commands" => [{
+              "description" => 'A nice method',
+              "action" => 'cool_command',
+              "name" => 'wazzzzzzaaa',
+              "regex" => 123,
+            }]
+          }
+        }
+
+        exception = assert_raises(Exceptions::InvalidHandler) do
+          Validator.verify_handler_integrity(handler_hash)
+        end
+        assert_equal(
+          "dummy_handler is not valid: [wazzzzzzaaa]: No regex was provided.",
+          exception.message
+        )
+      end
+
+      test "#verify_handler_integrity raises invalid base command error" do
+        handler_hash = {
+          "dummy_handler" => {
+            "commands" => [{
+              "description" => 'A nice method',
+              "action" => 'cool_command',
+              "name" => 'wazzzzzzaaa',
+              "base_command" => 123,
+            }]
+          }
+        }
+
+        exception = assert_raises(Exceptions::InvalidHandler) do
+          Validator.verify_handler_integrity(handler_hash)
+        end
+        assert_equal(
+          "dummy_handler is not valid: [wazzzzzzaaa]: No base command was provided.",
```

Comment:
A base command was provided here, but of the wrong type

Reply:
Whoops, forgot to change the error message updated!